rtl/src/main/scala/tensil/tcu/Sampler.scala

```diff
@@ -74,25 +67,23 @@ class Sampler(blockSize: Int) extends Module {
     sample.programCounter := io.programCounter
     sample.flags := io.flags
 
-    when(cycleCounter === (io.sampleInterval - 1.U)) {
+    when(cycleCounter === 0.U) {
       sampleReady := true.B
 
-      cycleCounter := 0.U
+      cycleCounter := (io.sampleInterval - 1.U)
     }.otherwise {
       sampleReady := false.B
 
-      cycleCounter := cycleCounter + 1.U
+      cycleCounter := cycleCounter - 1.U
     }
   }.otherwise {
-    sample.programCounter := invalidProgramCounter
-    sample.flags := invalidFlags
-
-    sampleReady := true.B
+    sample := WireDefault(new Sample(), 0.U.asTypeOf(new Sample()))
```

Comment:
Does the driver need to change to incorporate the new invalid value?

Reply:
Yes, the driver changes will follow up in separate PR.